ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -233,7 +233,7 @@ let get_initial_guest_metrics (lookup: string -> string option) (list: string ->
   (* to avoid breakage whilst 'micro' is added to linux and windows agents, default this field
      to -1 if it's not present in xenstore *)
   let pv_drivers_version =
-    if List.mem_assoc "micro" pv_drivers_version then pv_drivers_version (* already there; do nothing *)
+    if List.length pv_drivers_version = 0 || List.mem_assoc "micro" pv_drivers_version then pv_drivers_version (* already there; do nothing *)
```

Comment:
Minor point: `if pv_drivers_version = []` would be better to check for an empty list.